projects/plugins/crm/api/create_transaction.php

```diff
@@ -107,57 +125,33 @@
 				);
 
 }
-
-		// RECOMMENDED
-		$status = '';
-if ( isset( $new_trans['status'] ) ) {
-	$status = sanitize_text_field( $new_trans['status'] );
-}
-		$total = '';
-if ( isset( $new_trans['total'] ) ) {
-	$total = sanitize_text_field( $new_trans['total'] );
-}
-		$item_title = '';
-if ( isset( $new_trans['item_title'] ) ) {
-	$item_title = sanitize_text_field( $new_trans['item_title'] );
-}
-		$net = '';
-if ( isset( $new_trans['net'] ) ) {
-	$net = sanitize_text_field( $new_trans['net'] );
-}
-		$tax = '';
-if ( isset( $new_trans['tax'] ) ) {
-	$tax = sanitize_text_field( $new_trans['tax'] );
-}
-		$fee = '';
-if ( isset( $new_trans['fee'] ) ) {
-	$fee = sanitize_text_field( $new_trans['fee'] );
-}
-		$disc = '';
-if ( isset( $new_trans['disc'] ) ) {
-	$disc = sanitize_text_field( $new_trans['discount'] );
-}
-		$rate = '';
-if ( isset( $new_trans['rate'] ) ) {
-	$rate = sanitize_text_field( $new_trans['tax'] );
-}
-		$date = isset( $new_trans['date'] ) ? date( 'Y-m-d H:i:s', (int) $new_trans['date'] ) : '';
-
-		$tFields = array(
+		$transaction_fields = array(
 			'orderid'  => $orderid,
 			'customer' => $customer,
-			'status'   => $status,
-			'total'    => $total,
-			'date'     => $date,
-
-			'item'     => $item_title,
-			'net'      => $net,
-			'tax'      => $tax,
-			'fee'      => $fee,
-			'discount' => $disc,
-			'tax_rate' => $rate,
+			'status'   => '',
+			'total'    => '',
+			'date'     => '',
+			'item'     => '',
+			'net'      => '',
+			'tax'      => '',
+			'fee'      => '',
+			'discount' => '',
+			'tax_rate' => '',
 		);
 
+		foreach ( $new_trans as $field => $value ) {
+
+			if ( $field === 'orderid' || $field === 'email' || $field === 'fname' ) {
```

Comment:
Why get the `orderid` and `fname` separately above rather than processing them here? I would suggest handling "required" blank fields after processing them.

Reply:
- `fname` is required above, like `email`, to be able to create the contact if it does not exist. They are not part of the transaction object.
- `orderid` is not part of the transaction object. See `addOrUpdateTransaction`. We should use `ref` instead of. (DALv3). Updated.